tarxx.h

```diff
@@ -101,20 +104,102 @@ namespace tarxx {
+
             block_t block;
             std::fstream infile(filename, std::ios::in|std::ios::binary);
             if (!infile.is_open()) return;
             write_header(filename);
             while (infile.good()) {
                 infile.read(block.data(), block.size());
-                auto read = infile.gcount();
+                const auto read = infile.gcount();
                 if (read < block.size()) std::fill_n(block.begin() + read, block.size() - read, 0);
                 write(block);
             }
         }
 
+        void add_file_streaming()
+        {
+            if (stream_file_header_pos_ >= 0) throw std::logic_error("Can't add new file before add_file_complete has been called");
+            if (mode_ != output_mode::file_output) throw std::logic_error(__func__ + " only supports output mode file"s);
+
+            // write emtpy header
+            stream_file_header_pos_ = file_.tellg();
+            block_t header {};
+            write(header);
+        }
+
+        void add_file_streaming_data(const char* const data, std::streamsize size)
+        {
+            unsigned long pos = 0;
+
+            // fill a new block with old and new data if enough data
+            // is available to fill a whole block
+            if ((stream_block_used_ + size) >= BLOCK_SIZE) {
+                block_t block;
+
+                // copy saved data
+                std::copy_n(stream_block_.data(), stream_block_used_, block.data());
+
+                // copy from new data
+                const auto copy_from_new_data = BLOCK_SIZE - stream_block_used_;
+                std::copy_n(data, copy_from_new_data, block.data() + stream_block_used_);
+
+                // write new complete block
+                write(block);
+
+                pos += copy_from_new_data;
+                size -= copy_from_new_data;
+                stream_block_used_ = 0;
+            }
+
+            // write new block as long as we have enough data to fill a whole block
+            while (size >= BLOCK_SIZE) {
+                block_t block;
+                std::copy_n(data + pos, BLOCK_SIZE, block.data());
+                pos += BLOCK_SIZE;
+                size -= BLOCK_SIZE;
+                write(block);
+            }
+
+            // store remaining data for next call or stream_file_complete
+            std::copy_n(data + pos, size, stream_block_.data() + stream_block_used_);
+            stream_block_used_ += size;
+        }
+
+        void stream_file_complete(const std::string& filename, __mode_t mode, __uid_t uid, __gid_t gid, __off_t size, __time_t mod_time)
+        {
+            // create last block and fill with 0 if necessary
+            block_t block;
+            if (stream_block_used_ < block.size())
+                std::fill_n(block.begin() + stream_block_used_, block.size() - stream_block_used_, 0);
```

Comment:
good you thought about this case. I'd rather have it zero-intiliazed instead of `fill_n`-ing it, like we do elsewhere already.

```suggestion
            block_t block {};
```

Reply:
makes sense :+1: